ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessorIntegrationTest.java

```diff
@@ -0,0 +1,759 @@
+    Address contractAddress = Address.fromHexStringStrict(CONTRACT_ADDRESS);
+
+    Transaction getSlot1Transaction =
+        createContractUpdateSlotTransaction(
+            0, contractAddress, "getSlot1", keyPair1, Optional.empty());
+    Transaction setSlot1Transaction =
+        createContractUpdateSlotTransaction(
+            0, contractAddress, "setSlot1", keyPair2, Optional.of(1000));
+    Transaction setSlo2Transaction =
+        createContractUpdateSlotTransaction(
+            1, contractAddress, "setSlot2", keyPair1, Optional.of(2000));
+    Transaction setSlot3Transaction =
+        createContractUpdateSlotTransaction(
+            2, contractAddress, "setSlot3", keyPair1, Optional.of(3000));
+
+    Block blockWithTransactions =
+        createBlockWithTransactions(
+            "0xdf21d4fef211d7a905022dc87f2a68f4bf9cb273fcf9745cfa7f7c2f258c03f3",
+            getSlot1Transaction,
+            setSlot1Transaction,
+            setSlo2Transaction,
+            setSlot3Transaction);
+    MutableWorldState worldState = worldStateArchive.getMutable();
+
+    BlockProcessingResult blockProcessingResult =
+        blockProcessor.processBlock(blockchain, worldState, blockWithTransactions);
+
+    assertTrue(blockProcessingResult.isSuccessful());
+
+    // Verify the state
+    assertContractStorage(worldState, contractAddress, 0, 1000);
+    assertContractStorage(worldState, contractAddress, 1, 2000);
+    assertContractStorage(worldState, contractAddress, 2, 3000);
+  }
+
+  void processAccountReadThenUpdateTx(final BlockProcessor blockProcessor) {
+    Address contractAddress = Address.fromHexStringStrict(CONTRACT_ADDRESS);
+    Transaction transactionTransfer = // ACCOUNT_GENESIS_1 -> CONTRACT_ADDRESS
+        createTransferTransaction(
+            0, 1_000_000_000_000_000_000L, 300000L, 5L, 7L, CONTRACT_ADDRESS, keyPair1);
+    Transaction getcontractBalanceTransaction =
+        createContractReadAccountTransaction(1, contractAddress, "getBalance", keyPair1, ACCOUNT_2);
+
+    Transaction sendEthFromContractTransaction =
+        createContractUpdateAccountTransaction(
+            0, contractAddress, "transferTo", keyPair2, ACCOUNT_2, 500_000_000_000_000_000L);
+
+    Block blockWithTransactions =
+        createBlockWithTransactions(
+            "0x91966cdde619acb05a1d9fef2f8801432a30edde7131f1f194002b0a766026c7",
+            transactionTransfer,
+            getcontractBalanceTransaction,
+            sendEthFromContractTransaction);
+    MutableWorldState worldState = worldStateArchive.getMutable();
+
+    BlockProcessingResult blockProcessingResult =
+        blockProcessor.processBlock(blockchain, worldState, blockWithTransactions);
+
+    assertTrue(blockProcessingResult.isSuccessful());
+
+    // Verify the state
+    BonsaiAccount contractAccount = (BonsaiAccount) worldState.get(contractAddress);
+    BonsaiAccount updatedAccount2 =
+        (BonsaiAccount) worldState.get(Address.fromHexStringStrict(ACCOUNT_2));
```

Comment:
This name is what I was expecting...but could be updateAccount0x2, would just be nice to have a consistent convention across the file.

Reply:
Yes, updated